### PR TITLE
DCD-289: Fix linting

### PR DESCRIPTION
**Pull request by nanux**
This one was slightly trickier.

Linting output before:
```
W2001 Parameter CustomDnsName not used.
templates/quickstart-bitbucket-dc-with-vpc.template.yaml:510:3

E3012 Property Resources/ClusterNodeGroup/Properties/DesiredCapacity/Fn::If/1 should be of type String
templates/quickstart-bitbucket-dc.template.yaml:662:7

E3012 Property Resources/ClusterNodeGroup/Properties/MinSize/Fn::If/1 should be of type String
templates/quickstart-bitbucket-dc.template.yaml:664:7

E2523 Only one of [VirtualName, Ebs, NoDevice] should be specified for Resources/ClusterNodeLaunchConfig/Properties/BlockDeviceMappings/0
templates/quickstart-bitbucket-dc.template.yaml:771:11

E3012 Property Resources/ClusterNodeScaleUpPolicy/Properties/Cooldown should be of type String
templates/quickstart-bitbucket-dc.template.yaml:799:7

E3012 Property Resources/ClusterNodeScaleDownPolicy/Properties/Cooldown should be of type String
templates/quickstart-bitbucket-dc.template.yaml:806:7

W1020 Fn::Sub isn't needed because there are no variables at Resources/FileServer/Metadata/AWS::CloudFormation::Init/config/commands/080_run_atl_init_node/command/Fn::Sub
templates/quickstart-bitbucket-dc.template.yaml:958:15

E2504 Iops shouldn't be defined for type gp2 for Resources/FileServer/Properties/BlockDeviceMappings/0/Ebs/Iops
templates/quickstart-bitbucket-dc.template.yaml:967:13

E3012 Property Resources/FileServer/Properties/NetworkInterfaces/0/DeviceIndex should be of type String
templates/quickstart-bitbucket-dc.template.yaml:985:11

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is False and when condition "StandbyMode" is False for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is False and when condition "StandbyMode" is False for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is False for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is False for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E2521 Property MasterUserPassword should exist with MasterUsername when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1007:5

E3012 Property Resources/DB/Properties/EngineVersion should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1015:7

E2520 Property MasterUsername should NOT exist with SourceDBInstanceIdentifier when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1021:7

E2520 Property MasterUsername should NOT exist with SourceDBInstanceIdentifier when condition "RestoreFromRDSSnapshot" is False and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1021:7

E2520 Property MasterUsername should NOT exist with SourceDBInstanceIdentifier when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is False and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1021:7

E2520 Property MasterUsername should NOT exist with SourceDBInstanceIdentifier when condition "RestoreFromRDSSnapshot" is True and when condition "DBProvisionedIops" is True and when condition "SetDBMasterUserPassword" is False and when condition "RestoreRDSOrStandby" is True and when condition "StandbyMode" is True for Resources/DB/Properties
templates/quickstart-bitbucket-dc.template.yaml:1021:7

E3012 Property Resources/LoadBalancer/Properties/Listeners/0/LoadBalancerPort should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1044:11

E3012 Property Resources/LoadBalancer/Properties/Listeners/0/InstancePort/Fn::If/1 should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1046:11

E3012 Property Resources/LoadBalancer/Properties/Listeners/0/InstancePort/Fn::If/2 should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1046:11

E3012 Property Resources/LoadBalancer/Properties/Listeners/1/Fn::If/1/LoadBalancerPort should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1051:13

E3012 Property Resources/LoadBalancer/Properties/Listeners/1/Fn::If/1/InstancePort should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1053:13

E3012 Property Resources/LoadBalancer/Properties/Listeners/2/LoadBalancerPort should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1059:11

E3012 Property Resources/LoadBalancer/Properties/Listeners/2/InstancePort should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1061:11

E3012 Property Resources/LoadBalancer/Properties/HealthCheck/HealthyThreshold should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1064:9

E3012 Property Resources/LoadBalancer/Properties/HealthCheck/Interval should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1065:9

E3012 Property Resources/LoadBalancer/Properties/HealthCheck/Timeout should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1067:9

E3012 Property Resources/LoadBalancer/Properties/HealthCheck/UnhealthyThreshold should be of type String
templates/quickstart-bitbucket-dc.template.yaml:1068:9

E3012 Property Resources/SecurityGroupIngress/Properties/FromPort should be of type Integer
templates/quickstart-bitbucket-dc.template.yaml:1107:7

E3012 Property Resources/SecurityGroupIngress/Properties/ToPort should be of type Integer
templates/quickstart-bitbucket-dc.template.yaml:1108:7
```

Build
https://server-syd-bamboo.internal.atlassian.com/browse/ZDCD-AWSBITBUCKET4-LINT-3

### CODE REVIEW

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 1]
```diff
@@ -955,19 +955,19 @@ Resources:
...
               ignoreErrors: true
 
     Properties:
       BlockDeviceMappings:
         - DeviceName: /dev/sdf
           Ebs:
-            VolumeType: !If [IsHomeProvisionedIops, io1, gp2]
-            Iops: !If [IsHomeProvisionedIops, !Ref HomeIops, !Ref "AWS::NoValue"]
             DeleteOnTermination: !Ref HomeDeleteOnTermination
+            Iops: !If [IsHomeProvisionedIops, !Ref HomeIops, !Ref "AWS::NoValue"]
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc.template.yaml:966]: This is the reason why we are ignoring `E2504` check for now. The rule currently cannot handle `Iops` and `VolumeType` when the condition is used for `If` function.

File: templates/quickstart-bitbucket-dc-with-vpc.template.yaml [hunk 2]
```diff
@@ -566,6 +566,7 @@ Resources:
         ClusterNodeMax: !Ref 'ClusterNodeMax'
         ClusterNodeMin: !Ref 'ClusterNodeMin'
         CreateBucket: !Ref 'CreateBucket'
+        CustomDnsName: !Ref 'CustomDnsName'
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc-with-vpc.template.yaml:569]: This was a bug - even when user set `CustomDnsName` parameter in the master template it wasn't propagated.

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 1 above)
Comment (bpar476) [reply templates/quickstart-bitbucket-dc.template.yaml:966]: Rough, can we raise an issue with the project?

**Review by bpar476 — APPROVED** [review]
clean and good to catch those bugs 👏

**Review by tarka — APPROVED** [review]

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 1 above)
Comment (nanux) [reply templates/quickstart-bitbucket-dc.template.yaml:966]: 🖖, good thinking. I had it in my todo but it was too late yesterday to start raising issues in public projects.

**Pull request merged**